src/main/java/dinkplugin/notifiers/GambleNotifier.java

```diff
@@ -0,0 +1,138 @@
+
+    @Value
+    @VisibleForTesting
+    static class ParsedData {
+        @NonNull
+        String itemName;
+        int itemQuantity;
+        @Nullable
+        String tertiaryItem;
+        int gambleCount;
+
+        Stream<Pair<String, Integer>> itemsWithQuantities() {
+            return Stream.of(Pair.of(itemName, itemQuantity), Pair.of(tertiaryItem, 1))
+                .filter(p -> p.getLeft() != null);
+        }
+    }
+
+    @Inject
+    private ItemSearcher itemSearcher;
+
+    @Inject
+    private ItemManager itemManager;
+
+    @Override
+    public boolean isEnabled() {
+        return config.notifyGamble() && super.isEnabled();
+    }
+
+    @Override
+    protected String getWebhookUrl() {
+        return config.gambleWebhook();
+    }
+
+    public void onMesBoxNotification(String message) {
+        if (!isEnabled()) return;
+        ParsedData data = parse(message);
+        if (data != null) {
+            handleNotify(data);
+        }
+    }
+
+    @Nullable
+    @VisibleForTesting
+    static ParsedData parse(String message) {
+        Matcher matcher = GAMBLE_REGEX.matcher(message);
+        if (!matcher.matches()) return null;
+        String rawItem = matcher.group(1).trim();
+        String itemName = rawItem;
+        int itemQuantity = 1;
+        Matcher quantityMatcher = ITEM_QUANTITY_REGEX.matcher(rawItem);
+        if (quantityMatcher.matches()) {
+            itemName = quantityMatcher.group(1).trim();
+            itemQuantity = Integer.parseInt(quantityMatcher.group(2));
+        }
+        String tertiary = matcher.group(2);
+        if (tertiary != null) {
+            tertiary = tertiary.replaceAll("!", "").trim();
+        }
+        int gambleCount = Integer.parseInt(matcher.group(3));
+        return new ParsedData(itemName, itemQuantity, tertiary, gambleCount);
+    }
+
+    private void handleNotify(ParsedData data) {
+        String messageFormat;
+        if (config.gambleRareLoot() && RARE_LOOT.contains(data.itemName.toLowerCase())) {
+            messageFormat = config.gambleRareNotifyMessage();
+        } else if (data.gambleCount % config.gambleInterval() == 0) {
+            messageFormat = config.gambleNotifyMessage();
+        } else {
+            return;
+        }
+        String player = Utils.getPlayerName(client);
+        String message = StringUtils.replaceEach(
+            messageFormat,
+            new String[] { "%USERNAME%", "%COUNT%", "%LOOT%" },
+            new String[] { player, String.valueOf(data.gambleCount), lootSummary(data) }
+        );
+        createMessage(config.gambleSendImage(), NotificationBody.builder()
+            .text(message)
+            .extra(new GambleNotificationData(data.gambleCount, serializeItems(data)))
+            .type(NotificationType.BARBARIAN_ASSAULT_GAMBLE)
+            .build());
+    }
+
+    // ex. "watermelon seed x 50, clue scroll (elite)"
+    private String lootSummary(ParsedData data) {
```

Comment:
With this converted to accept `SerializedItemStack`s, we can stringify items with the same format as clue/loot notifier: https://github.com/pajlads/DinkPlugin/blob/master/src/main/java/dinkplugin/notifiers/ClueNotifier.java#L136 (with new-line as delimeter)

Reply:
Was looking for a util function for this. Maybe it's worth adding it to `ItemUtils`?

Reply:
good idea 👍